doc/BGP/BGP-supress-fib-pending.md

```diff
@@ -0,0 +1,804 @@
+    syncd -. "SAIRedis Bulk Reply" -.-> RouteOrch
+```
+
+A ```ResponsePublisher``` must have a constructor that accepts a ```RedisPipeline``` and a flag ```buffered``` to make it use the pipelining. The constructor is similar to one in use with ```ProducerStateTable```:
+
+```c++
+ResponsePublisher::ResponsePublisher(bool buffered = false);
+```
+
+And an API to set buffered mode:
+
+```c++
+void ResponsePublisher::setBuffered(bool buffered);
+```
+
+The ```OrchDaemon``` has to flush the ```RedisPipeline``` in ```OrchDaemon::flush``` when there are no pending tasks to perform.
+
+#### 7.6. Consistency monitoring script and mitigation action
+
+Routing is a crucial part of a network switch. This feature adds additional flows in existing route processing pipeline and so along the way there might be unexpected failures leading to routes being not marked as offloaded in zebra - missed notification, race conditions leading to in-consistency, etc. It is required to monitor the consistency of routes state periodically and mitigate problems.
+
+At the moment *route_check.py* verifies routes between APPL_DB & ASIC_DB and makes sure they are in sync. In addition to that it is required to extend this script to check for consistency APPL_DB, APPL_STATE_DB and zebra by calling ```show ip route json``` and verify every route installed in ASIC_DB is marked as offloaded. The script will retry the check for 3 times every 15 sec and if zebra FIB is not in sync with ASIC_DB a mitigation action is performed. The mitigation action publishes required notifications to trigger fpmsyncd flows to send RTM_NEWROUTE message to zebra and log alerts in the syslog.
+
+This way the problem is mitigated automatically and a network operator is notified via syslog alert about a problem.
+
+### 8. SAI API
+
+No new SAI API or changes to SAI design and behavior needed for this functionality.
+
+### 9. Warmboot and Fastboot Design Impact
+
+#### 9.1. Warm Reboot
+
+With BGP Graceful Restart, peers are keeping advertised routes in the FIB while the switch restarts.
+When fpmsyncd is about to set *reconciled* state it must send RTM_NEWROUTE for all prefixes that were restored previously from APPL_DB since they are already in the ASIC. Once all existing routes are pushed to zebra a comparison logic will finish the warm reboot process and fpmsyncd will continue in a normal operational mode.
+
+#### 9.2. Fast Reboot
+
+During fast reboot BGP session is closed by SONiC device without the notification. BGP session is preserved in graceful restart mode. BGP routes on the peer are still active, because nexthop interfaces are up. Once interfaces go down, received BGP routes on the peer are removed from the routing table. Nothing is sent to SONiC device since then. After interfaces go up and BGP sessions re-establish the peer's BGP re-learns advertised routes.
+
+Due to additional response publishing in orchagent there might be a slight delay in fast reboot reconfiguration.
```

Comment:
What's the fast boot time with this change?

Reply:
Must be less than 30 sec down time